lib/cylc/scheduler.py

```diff
@@ -1691,9 +1597,8 @@ def shutdown(self, reason=None):
                 self.task_job_mgr.unlink_hosts_contacts(self.suite)
 
         # disconnect from suite-db, stop db queue
-        if getattr(self, "db", None) is not None:
-            self.pri_dao.close()
-            self.pub_dao.close()
+        if getattr(self, "suite_db_mgr", None) is not None:
```

Comment:
Actually, I lied. The attribute is always set to a value, so I think I can get rid of this test.